src/main/java/com/hubspot/jinjava/lib/tag/IncludeTag.java

```diff
@@ -106,7 +106,7 @@ public String interpret(TagNode tagNode, JinjavaInterpreter interpreter) {
         .getCurrentPathStack()
         .push(templateFile, interpreter.getLineNumber(), interpreter.getPosition());
 
-      return interpreter.render(node);
+      return interpreter.render(node, false);
```

Comment:
could you explain a bit more why this is more correct?

Reply:
I’ve been trying to check other examples of Cycle Exception being thrown. It was the only combination of tags like “import/export/extends” that reacted like that, ending up with TagCycleException which seems some kind of last resort in the case when common types of exceptions weren’t handled.
In that instance, we have an error because we went to render() method again before we actually pop the node from CurrentPathStack.

Comparing to “import”: there we also try to render the template when we got to the imported part, but unlike “include” we’re creating a new interpreter instance. It’s alright because jinja documentation told us so: “It’s important to know that imports are cached and imported templates don’t have access to the current template variables”
However, we cannot use the same trick with “import”.  About this tag jinja has that statement: “Included templates have access to the variables of the active context by default”. So we must use the same interpreter for both the original templates and the imported ones.

Knowing that I found the only way to get rid of that problem: forbid the included template to render whatever we have inside ExtendedRoots. It shouldn’t break anything because eventually we will process them but it’s better to do that outside the imported template.

Sources for the behavior of each tag:
https://jinja.palletsprojects.com/en/2.11.x/templates/#include
https://jinja.palletsprojects.com/en/2.11.x/templates/#import